src/crypto/hsm/CHIPCryptoPALHsm_config.h

```diff
@@ -60,6 +60,8 @@
 #define ENABLE_HSM_EC_KEY
 #define ENABLE_HSM_CASE_EPHEMERAL_KEY
 #define ENABLE_HSM_CASE_OPS_KEY
+/* Provision HSM with device attestation key before enabling ENABLE_HSM_DEVICE_ATTESTATION */
+//#define ENABLE_HSM_DEVICE_ATTESTATION
```

Comment:
When HSM is enabled, it can be used to delegate other crypto operations also. Only if the DAC key is provisioned in HSM, this macro can be enabled to use HSM for device attestation phase.

Reply:
Yes. My point is that the choice to use the ENABLE_HSM_DEVICE_ATTESTATION should be permanent for a particular project. You should not need to edit this file in core SDK. There is no way to override this.

You need to make sure that this is overridable on a per-product basis.

Reply:
Added new build arg to enable HSM for device attestation